c2corg_ui/format/toc.py

```diff
@@ -0,0 +1,82 @@
+"""
+Table of Contents Extension for Python-Markdown
+===============================================
+
+See <https://pythonhosted.org/Markdown/extensions/toc.html>
+for documentation.
+
+Oringinal code Copyright 2008 [Jack Miller](http://codezen.org)
+
+All changes Copyright 2008-2014 The Python Markdown Project
+
+License: [BSD](http://www.opensource.org/licenses/bsd-license.php)
+
+Modified for C2C : remove title emphasis, and only include <4 levels
+"""
+
+from markdown.extensions.toc import (TocExtension, TocTreeprocessor,
+                                     string_type, stashedHTML2text,
+                                     unique, nest_toc_tokens)
+
+
+class C2CTocTreeprocessor(TocTreeprocessor):
+    def run(self, doc):
+
+        def not_emphasis(elt):
+            return elt.attrib.get("class") != "header-emphasis"
+
+        # Get a list of id attributes
+        used_ids = set()
+        for el in doc.iter():
+            if "id" in el.attrib:
+                used_ids.add(el.attrib["id"])
+
+        toc_tokens = []
+        for el in doc.iter():
+            if isinstance(el.tag, string_type) and \
+                    self.header_rgx.match(el.tag):
+                self.set_level(el)
+
+                # modification for camptocamp
```

Comment:
Shouldn't we instead simply point to the file you modified on github, linking with the revision that was used?

Reply:
Actually, I've tried to modify less code of official extension as possible. Unfortunalty key method run() was quite big. 

As I need to modify only two lines, I've commented them and what I've done for future maintenance.

Reply:
I understand that. What I was suggessting is that pointing to the original source file would be more readable. But it's not important.

Reply:
there is a header on top of the file with all mention, is it what you suggest?

Reply:
Non. Nevermind. 😉